### ISSUE TITLE
no-restyle does not trace descendant-selector restyles (`[&_Child]:bg-*`) back to design-system components

### ISSUE THREAD

**Issue by emirhangungormez** [shadcn-ui/lint#5]
Title: no-restyle does not trace descendant-selector restyles (`[&_Child]:bg-*`) back to design-system components

## Problem

`no-restyle` enforces a component's appearance contract when classes are applied directly to the component. It does not trace an arbitrary-variant descendant selector on a parent element down to a wrapped child component:

​```tsx
import { Button } from "@/components/ui/button"

export function Save() {
  // no-restyle: no finding. Button's color contract is silently bypassed.
  return (
    <div className="[&>button]:bg-pink-500 [&>button]:rounded-full">
      <Button>Save</Button>
    </div>
  )
}
​```

`docs/how-it-works.md` already lists this under "What it cannot see" ("Parent selectors... The same applies to `*:` and `**:`. Token rules still check these classes."), so the gap is documented. Two things push it from "acknowledged limitation" toward "worth a tracking issue":

1. **It's a category bypass, not just an edge case.** `no-arbitrary-values` and `no-raw-colors` already parse `[&_svg]:size-4`-style selectors and validate the *value* inside them. Only `no-restyle`'s *ownership* check (is this component even allowed to have its color/shape/spacing touched?) skips them. An agent can pass every configured rule while fully repainting `Button`, `Card`, etc. from outside.
2. **The rule's own suggested fix points here.** The `layoutClass` message tells an agent to put disallowed layout classes on a parent element. Nothing stops the same agent from reaching for `[&_Button]:` on that parent for an *appearance* class once direct restyling is blocked — exactly the kind of workaround `docs/evals.md`'s red-team section measures for seven other escapes. This one isn't in that set.

## Why it matters

The README's pitch is "you decide what can change" via contracts. If a selector-based path around the contract isn't checked, the contract only holds as long as the agent restyles the component directly — which undercuts the guarantee for exactly the audience (an autonomous agent looking for the shortest path to zero errors) this project targets.

## Suggested scope for a first pass

Given the existing "one hop" precedent for class-value tracing and wrapper forwarding (see `how-it-works.md#wrappers` and `#values-in-variables`), a bounded v1 seems more tractable than solving the general CSS cascade problem:

- Same file only, JSX-local: when an ancestor element's `className` contains an arbitrary-variant selector (`[&_X]`, `[&>X]`, `*:`, `**:`) whose selector matches a recognized design-system component rendered as a descendant in the same return, attribute the appearance classes inside that selector to that component's contract.
- Consider warn-level or opt-in at first, since matching against arbitrary CSS selectors (tag names, classes, attribute selectors) is fuzzier than the exact-import resolution the rest of the project relies on.
- At minimum, add this pattern to `packages/evals`'s red-team suite so it's tracked like the other seven escapes, even before detection lands.

Happy to take a pass at the red-team-suite addition and/or a first detection implementation if this is a direction you'd want — wanted to check scope and appetite first, since the selector-matching involved may deserve a design discussion before a PR.
